repair_order_group/models/repair_order.py

```diff
@@ -0,0 +1,248 @@
+            return super().action_repair_cancel()
+
+        # Collect ALL repairs to cancel: selected + their group members
+        group_repairs = self.mapped("grouped_repair_ids").filtered(
+            lambda r: r.state not in ("cancel", "done")
+        )
+        all_repairs = (self | group_repairs).with_context(skip_group_cancel=True)
+
+        # Prevent calling the original method on an empty recordset
+        if not all_repairs:
+            return True
+
+        # Single call to original method to preserve full Odoo workflow
+        return super(RepairOrder, all_repairs).action_repair_cancel()
+
+    def action_create_sale_order(self):
+        """Create Sale Order - single one for group or standard behavior if no group.
+
+        For repair groups: creates one sale order containing all valid repairs
+        from the same group. Maintains core Odoo validations while extending
+        functionality for grouped repairs.
+
+        Returns:
+            dict: Action to view the created sale order
+        """
+
+        # Core Odoo validations
+        if any(repair.sale_order_id for repair in self):
+            concerned = self.filtered("sale_order_id")
+            ref_str = "\n".join(ro.name for ro in concerned)
+            raise UserError(
+                _(
+                    "You cannot create a quotation for a repair order that is "
+                    "already linked to an existing sale order.\n"
+                    "Concerned repair order(s):\n%(ref_str)s",
+                    ref_str=ref_str,
+                )
+            )
+
+        if any(not repair.partner_id for repair in self):
+            concerned = self.filtered(lambda ro: not ro.partner_id)
+            ref_str = "\n".join(ro.name for ro in concerned)
+            raise UserError(
+                _(
+                    "You need to define a customer for a repair order in order to "
+                    "create an associated quotation.\n"
+                    "Concerned repair order(s):\n%(ref_str)s",
+                    ref_str=ref_str,
+                )
+            )
+
+        # Group logic only if repairs belong to ONE single group
+        grouped = self.filtered(lambda r: r.group_id)
+        if grouped:
+            groups = grouped.mapped("group_id")
+
+            # If repairs belong to 0 or >1 groups → fallback to default behavior
+            if len(groups) != 1:
+                return super().action_create_sale_order()
+
+            group = groups[0]
+
+            # Group-specific validations
+            if group.repair_ids.filtered("sale_order_id"):
+                raise UserError(_("A sale order already exists for this repair group."))
+
+            repairs = group.repair_ids.filtered(
+                lambda r: not r.sale_order_id and r.state != "cancel"
+            )
+            if not repairs:
+                raise UserError(_("No valid repairs found in this group."))
```

Comment:
You select 1000 repairs, click "Create Quotation". On of the repairs fail. How can you know which repair causes the problem?